roles/neutron-data/tasks/main.yml

```diff
@@ -49,6 +49,18 @@
     - ursula_os == 'rhel'
     - neutron.plugin == 'ml2'
 
+- name: mask unwanted neutron services on compute (rhel)
+  command: systemctl mask {{ item.name }}
+  with_items:
+    - "{{ neutron.services.neutron_server }}"
+    - "{{ neutron.services.neutron_dhcp_agent }}"
+    - "{{ neutron.services.neutron_l3_agent }}"
+    - "{{ neutron.services.neutron_metadata_agent }}"
+  when:
+    - ursula_os == 'rhel'
+    - inventory_hostname in groups['compute']
```

Comment:
`inventory_hostname` is never in `groups[controller]` because `neutron-data` only runs on compute and network hosts. In that case, we can just check for not `network` hosts.
```
  when:
    - ursula_os == 'rhel'
    - not inventory_hostname in groups['network']
```
Bottomline is that we need to consider the converged controllers scenario. However checking for just not `network` hosts will allow us to separate `network` nodes from `controller` nodes easily in the future.